dpcpp/base/helper.hpp

```diff
@@ -84,10 +86,13 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
                       sycl::queue* queue, InferredArgs... args)       \
     {                                                                 \
         queue->submit([&](sycl::handler& cgh) {                       \
-            cgh.parallel_for(sycl_nd_range(grid, block),              \
-                             [=](sycl::nd_item<3> item_ct1) {         \
-                                 kernel_<encoded>(args..., item_ct1); \
-                             });                                      \
+            cgh.parallel_for(                                         \
+                sycl_nd_range(grid, block), [=                        \
+            ](sycl::nd_item<3> item_ct1) [[sycl::reqd_sub_group_size( \
+                                                KCFG_1D::decode<1>(   \
```

Comment:
this is not always KCFG_1D.
and it will give error when the subgroup size is equal to 1, I think

Reply:
Could you add a int map for it? only specialize the 1 to 16(the default of gpu). others just return themselves